File: typescript/word-count/word-count.ts
export function count(subtitle: string): Map<string, number> {
  const words = subtitle.toLowerCase().match(/\b[a-zA-Z0-9]+('[a-zA-Z0-9]+)?\b/g);

  const wordCountMap = new Map<string, number>();
  if (words) {
    for (const word of words) {
      const count = wordCountMap.get(word) || 0;
      wordCountMap.set(word, count + 1);
    }
  }
  return wordCountMap;
}6

File: go/word-count/word_count.go
package wordcount

import (
	"regexp"
	"strings"
)

type Frequency map[string]int

func WordCount(str string) Frequency {
	str = strings.ToLower(str)
	re := regexp.MustCompile(`\b[a-zA-Z0-9]+('[a-zA-Z0-9]+)?\b`)
	words := re.FindAllString(str, -1)
	wordCountMap := make(map[string]int)
	for _, word := range words {
		wordCountMap[word]++
    }
	return wordCountMap
}